pubtools/exodus/_tasks/push.py

```diff
@@ -30,32 +29,40 @@ def run(self):
         publish_id = str(publish.get("id"))
         LOG.info("Publish ID: %s", publish_id)
 
-        for item in self.args.sync:
-            LOG.debug("Processing %s", item)
-            items = item.split(",")
-            src, dest = items[0].strip(), items[1].strip()
-            dest = dest if ":" in dest else "exodus:%s" % dest
-
-            cmd = ["exodus-rsync", "--exodus-publish", publish_id, src, dest]
-            if self.args.verbose:
-                cmd.append("-" + "v" * self.args.verbose)
-            if self.extra_args:
-                cmd.extend(self.extra_args)
-
-            LOG.info(" ".join(cmd))
-
-            proc = subprocess.Popen(
-                cmd,
-                stdout=subprocess.PIPE,
-                stderr=subprocess.STDOUT,
-                universal_newlines=True,
-            )
-            for line in proc.stdout:
-                LOG.info(line.strip())
-
-            ret = proc.wait()
-            if ret != 0:
-                raise RuntimeError("Exodus push failed")
+        with Source.get(self.args.source) as source:
+            for item in source:
+                if item.src and len(item.dest) == 1:
```

Comment:
Minor: if you feel like this is getting a bit too nested (which I personally do), you can always put the iteration logic in its own method something like this:

```python
@property
def push_items(self):
    with Source.get(self.args.source) as source:
        for item in source:
            if item.src and len(item.dest) == 1:
                yield item
            else:
                LOG.warning(...)
```

The main loop here would then look cleaner like:

```python
for item in self.push_items:
    LOG.debug("Processing %s", item)
    # ...the stuff for running exodus-rsync
```